File: eShopApi/Models/Product.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace eShopApi.Models
{
    public class Product
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; } = "";
        [BsonElement]
        public string Name { get; set; } = "";

        [BsonElement]
        public double Price { get; set; } = 0;
        [BsonElement]
        [BsonRepresentation(BsonType.ObjectId)]
        public string CategoryId { get; set; } = "";

        [BsonElement]
        [BsonRepresentation(BsonType.ObjectId)]
        public string SubCategoryId { get; set; } = "";

        [BsonElement]
        [BsonRepresentation(BsonType.ObjectId)]
        public string ShopId { get; set; } = "";

        [BsonElement]
        [BsonRepresentation(BsonType.ObjectId)]
        public string CreatedBy { get; set; } = "";

        [BsonElement]
        public List<string> Tags { get; set; } = new List<string>();

        [BsonElement]
        public Boolean IsHotItem { get; set; } = false;

        [BsonElement]
        public Boolean IsActive { get; set; } = true;

        [BsonElement]
        public Boolean OutOfStoke { get; set; } = false;

        [BsonElement]
        public List<ImagesProduct> Images { get; set; } = new List<ImagesProduct>();
        [BsonElement]
        public DateTime CreatedOn { get; set; } = DateTime.Now;
        [BsonElement]
        public DateTime ModifiedOn { get; set; } = DateTime.Now;
    }

    public class ImagesProduct
    {
        [BsonElement]
        public Boolean Main { get; set; } = false;

        [BsonElement]
        public string Url { get; set; } = "";
    }
}

File: eShopFrontend/src/interface/product.ts
export interface IProduct {
  id: string
  name: string
  price: number
  tags: string[]
  images: IProductImage[]
  categoryId: string
  categoryName: string
  subCategoryId: string
  subCategoryName: string
  shopId: string
  shopName: string
  createdBy: string
  userName: string
  isHotItem: boolean
}

export interface IProductImage {
  main: boolean
  url: string
}

export interface ICProduct {
  name: string
  price: number
  tags: string[]
  fileInfo: ICProductImage[]
  categoryId: string
  subCategoryId: string
  shopId: string
}

export interface ICProductImage {
  main: boolean
  file: File
}